### ISSUE TITLE
Need dict for build options in CIME build - SPA entry for scream_input.yaml is hardcoded to ne4

### ISSUE THREAD

**Issue by AaronDonahue** [E3SM-Project/scream#1421]
We currently don't have a way to designate which SPA remap file to use depending on the grid resolution of the run.  So we have hardcoded the `scream_input.yaml` file to match what is needed for an ne4 run.

We need a way at build time to designate the appropriate SPA remap file.  This issue focuses on SPA, but the same functionality will come in handy for a number of unique configurations.  For example, for running different homme namelist options, or different initial condition files.

EAM currently uses XML.  We could follow EAMs example or go a different route.

May be related to #1420

**Comment by bartgol** [comment]
I don't think this is related to #1420. But it is an important task. I think it should boil down to mods to our `buildnml` python script, which can access CASE configurations, and use them accordingly when generating namelists/yaml files.

I suspect this might be more tedious than hard. We could start with adding support for one or two CASE-related variables.

**Comment by jgfouca** [comment]
I believe this has been solved by the new XML namelist system.
